src/pykeen/models/base.py

```diff
@@ -547,6 +547,51 @@ def predict_scores_all_relations(
             scores = torch.sigmoid(scores)
         return scores
 
+    def score_t_inverse(self, hr_batch: torch.LongTensor, slice_size: Optional[int] = None):
+        r_inv_h = self._prepare_inverse_batch(batch=hr_batch, index=1)
+
+        if slice_size is None:
+            return self.score_h(rt_batch=r_inv_h)
+        else:
+            return self.score_h(rt_batch=r_inv_h, slice_size=slice_size)
+
+    def score_h_inverse(self, rt_batch: torch.LongTensor, slice_size: Optional[int] = None):
+        t_r_inv = self._prepare_inverse_batch(batch=rt_batch, index=0)
+
+        if slice_size is None:
+            return self.score_t(hr_batch=t_r_inv)
+        else:
+            return self.score_t(hr_batch=t_r_inv, slice_size=slice_size)
+
+    def score_hrt_inverse(
+        self,
+        hrt_batch: torch.LongTensor,
+    ) -> torch.FloatTensor:
+        r"""Score triples based on inverse triples, i.e., compute f(h,r,t) based on f(t,r_inv,h).
+
+        When training with inverse relations, the model produces two (different) scores for a triple $(h,r,t) \in K$.
+        This function enables users to inspect the scores obtained by using the corresponding inverse triples.
+        """
+        t_r_inv_h = self._prepare_inverse_batch(batch=hrt_batch, index=1)
+
+        return self.score_hrt(hrt_batch=t_r_inv_h)
+
+    def _prepare_inverse_batch(self, batch: torch.LongTensor, index_relation: int):
+        if not self.triples_factory.create_inverse_triples:
+            raise ValueError(
+                "Your model is not configured to predict with inverse relations."
+                " Set ``create_inverse_triples=True`` when creating the dataset/triples factory"
+                " or using the pipeline()."
+            )
+        batch_cloned = batch.clone()
+
+        # The number of relations stored in the triples factory includes the number of inverse relations
+        # Id of inverse relation: relation + 1
+        batch_cloned[:, index_relation] = batch_cloned[:, index_relation] + 1
+
+        # The score_t function requires (entity, relation) pairs instead of (relation, entity) pairs
+        return batch_cloned.flip(1)
+
     def predict_scores_all_heads(
```

Comment:
can you also improve the `predict_scores_all_tails()` function as well?

Reply:
Currently, we only expect the right side-side predictions, i.e., (h,r,*) to change its default behavior when the model has been trained with inverse relations (mainly because of the behavior of the LCWA training approach). That is why we don't need to adapt the `predict_scores_all_heads()` function. However, we have added in this PR the `score_h_inverse()`, which can be used to predict heads based on the inverse triples.

Reply:
Hold on, I think you made a typo - we adapted the `predict_scores_all_heads()` function but *not* the `predict_scores_all_tails()`. This is correct?

Reply:
Yes, good catch!

Reply:
💃 🦜